File: tests/fixtures/cpp/complex_method.cpp
int process_order(int status, int verified, int stock, int shipped, int cancelled, int delivered) {
    if (status == 0) {
        if (verified) {
            if (stock) {
                status = 1;
            } else {
                status = 2;
            }
        } else {
            status = 3;
        }
    } else if (status == 1) {
        if (shipped) {
            status = 4;
        } else if (cancelled) {
            status = 5;
        }
    } else if (status == 4) {
        if (delivered) {
            status = 6;
        }
    }
    return status;
}

File: tests/fixtures/kotlin/ComplexMethod.kt
class ComplexMethod {
    fun processOrder(
        status: Int,
        verified: Boolean,
        stock: Boolean,
        shipped: Boolean,
        cancelled: Boolean,
        delivered: Boolean
    ): String {
        var s = status
        if (s == 0) {
            if (verified) {
                if (stock) {
                    s = 1
                } else {
                    s = 2
                }
            } else {
                s = 3
            }
        } else if (s == 1) {
            if (shipped) {
                s = 4
            } else if (cancelled) {
                s = 5
            }
        } else if (s == 4) {
            if (delivered) {
                s = 6
            }
        }
        return s.toString()
    }
}
